RFC.md

```diff
@@ -0,0 +1,393 @@
+
+TODO: check if the above makes sense readily, or we want a
+better explanation or a better range.
+
+### Custom Tables
+
+Uses MAY create custom multihash tables to use custom hash
+functions not defined in the Standard Multihash Table. In
+order to avoid any ambiguity whatsoever, these custom tables
+MUST be strict super-sets of the Standard Multihash Table.
+Therefore, they MUST define any custom functions with the
+integer codes reserved for this purpose, and must use string
+names that begin with `x-`.
+
+  # example custom table. superset of SMT
+  # 0x01-0x0f reserved for application specific functions
+  # 0x10-0x3f reserved for SHA standard functions
+  0x00, identity
+  0x11, sha1
+  0x12, sha2-256
+  0x13, sha2-512
+  0x14, sha3-512
+  0x15, sha3-384
+  0x16, sha3-256
+  0x17, sha3-224
+  0x18, shake-128
+  0x19, shake-256
+  0x40, blake2b
+  0x41, blake2s
+
+### Self-Describing Multihash Tables
+
+It is possible to define a multihash table using completely
+self-describing definitions, meaning that the hash function
+itself is stored not by name but as a code value. How to do
+that is left up to future specifications. If one is defined,
+this specification SHOULD be updated to link to it.
+
+## Specification of the multihash format
+
+Multihash is a self-describing format to store, transmit, and
+display hash function digest values. It is unambiguous, provides
+"algorithmic agility", prevents "algorithm lock-in", improves the
+security of hash value transmissions, and improves the usability
+of hash functions.
+
+The multihash format is very simple. It defines a single value
+type: a multihash value. A multihash value consists of three parts:
+
+1. Hash Function Code: an integer code representing a hash
+  function from a pre-determined hash function table, expressed
+  as a varint.
+2. Digest Length: the length of the hash function digest value,
+  expressed as a varint.
+3. Digest Value: the hash function digest value, conforming to
+  the Digest Length
+
+These three parts are laid out in a byte sequence as follows:
+
+    <hash-function-code><digest-length><digest-value>
+
+The multihash format defines two representations for a multihash
+value:
+
+1. Packed Representation: a compact representation for use
+  on the wire, in storage, in displays, and in other identifiers.
+2. Explanation Representation: a larger, human readable
+  representation to make distinguishing values easier on
+  developers.
+
+Both representations follow the same parts order as above, only
+changing each of the parts.
+
+For all intents and purposes, the Packed Representation is the
+main contribution of this specification. The Explanation
+Representation is merely here to provide a single standard way to
+dump out values in more readable forms.
+
+### Packed Representation
+
+The Packed Representation is a compact representation optimized
+for transmitting, storing, displaying, and using multihashes.
```

Comment:
The packet representation isn't currently intended for display -- see comment under "explanation representation"